literal/utils.go

```diff
@@ -0,0 +1,131 @@
+package literal
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/substrait-io/substrait-go/expr"
+	"github.com/substrait-io/substrait-go/types"
+)
+
+func NewBool(value bool) (expr.Literal, error) {
+	return expr.NewPrimitiveLiteral[bool](value, false), nil
+}
+
+func NewInt8(value int8) (expr.Literal, error) {
+	return expr.NewPrimitiveLiteral[int8](value, false), nil
+}
+
+func NewInt16(value int16) (expr.Literal, error) {
+	return expr.NewPrimitiveLiteral[int16](value, false), nil
+}
+
+func NewInt32(value int32) (expr.Literal, error) {
+	return expr.NewPrimitiveLiteral[int32](value, false), nil
+}
+
+func NewInt64(value int64) (expr.Literal, error) {
+	return expr.NewPrimitiveLiteral[int64](value, false), nil
+}
+
+func NewFloat32(value float32) (expr.Literal, error) {
+	return expr.NewPrimitiveLiteral[float32](value, false), nil
+}
+
+func NewFloat64(value float64) (expr.Literal, error) {
+	return expr.NewPrimitiveLiteral[float64](value, false), nil
+}
+
+func NewString(value string) (expr.Literal, error) {
+	return expr.NewPrimitiveLiteral[string](value, false), nil
+}
+
+func NewDate(days int) (expr.Literal, error) {
+	return expr.NewLiteral[types.Date](types.Date(days), false)
+}
+
+func NewTime(tm time.Time) (expr.Literal, error) {
+	startOfTheDay := time.Date(tm.Year(), tm.Month(), tm.Day(), 0, 0, 0, 0, tm.Location())
+	return expr.NewLiteral[types.Time](types.Time(tm.Sub(startOfTheDay).Microseconds()), false)
+}
+
+func NewTimeFromMicros(micros int64) (expr.Literal, error) {
+	if micros < 0 || micros >= (24*time.Hour).Microseconds() {
+		return nil, fmt.Errorf("invalid time value %d", micros)
+	}
+	return expr.NewLiteral[types.Time](types.Time(micros), false)
+}
+
+func NewTimestamp(tm time.Time) (expr.Literal, error) {
```

Comment:
For these ones with conversions, we should clarify how conversion works in doc

Reply:
added doc